lib/validators/group_setting_validator.rb

```diff
@@ -0,0 +1,21 @@
+class GroupSettingValidator
+
+  include RegexSettingValidation
+
+  def initialize(opts = {})
+    @opts = opts
+    initialize_regex_opts(opts)
+  end
+
+  def valid_value?(val)
+    !val.present? || (Group.where(name: val).exists? && regex_match?(val))
```

Comment:
`val.blank?` is easier to grasp than `!val.present?`

Also, you can shorten the `exists?` call ;)

```suggestion
    val.blank? || (Group.exists?(name: val) && regex_match?(val))
```